src/views/forms/AdjustIsolatedMarginForm.tsx

```diff
@@ -0,0 +1,302 @@
+import { AlertType } from '@/constants/alerts';
+import { ButtonAction, ButtonShape } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+import { NumberSign, USD_DECIMALS } from '@/constants/numbers';
+
+import { useStringGetter } from '@/hooks';
+
+import { formMixins } from '@/styles/formMixins';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AlertMessage } from '@/components/AlertMessage';
+import { Button } from '@/components/Button';
+import { DiffOutput } from '@/components/DiffOutput';
+import { FormInput } from '@/components/FormInput';
+import { GradientCard } from '@/components/GradientCard';
+import { InputType } from '@/components/Input';
+import { OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
+
+import { getOpenPositionFromId, getSubaccount } from '@/state/accountSelectors';
+import { getMarketConfig } from '@/state/perpetualsSelectors';
+
+import { calculatePositionMargin } from '@/lib/tradeData';
+
+type ElementProps = {
+  marketId: SubaccountPosition['id'];
+};
+
+enum MarginAction {
+  ADD = 'ADD',
+  REMOVE = 'REMOVE',
+}
+
+const SIZE_PERCENT_OPTIONS = {
+  '5%': '0.05',
+  '10%': '0.1',
+  '25%': '0.25',
+  '50%': '0.5',
+  '75%': '0.75',
+};
+
+export const AdjustIsolatedMarginForm = ({ marketId }: ElementProps) => {
+  const stringGetter = useStringGetter();
+  const [marginAction, setMarginAction] = useState(MarginAction.ADD);
+  const subaccountPosition = useSelector(getOpenPositionFromId(marketId));
+  const { adjustedMmf, leverage, liquidationPrice, notionalTotal } = subaccountPosition ?? {};
+  const marketConfig = useSelector(getMarketConfig(marketId));
+  const { tickSizeDecimals } = marketConfig ?? {};
+
+  /**
+   * @todo: Replace with Abacus functionality
+   */
+  const [percent, setPercent] = useState('');
+  const [amount, setAmount] = useState('');
+  const onSubmit = () => {};
+
+  const positionMargin = {
+    current: calculatePositionMargin({
+      adjustedMmf: adjustedMmf?.current,
+      notionalTotal: notionalTotal?.current,
+    }).toFixed(tickSizeDecimals ?? USD_DECIMALS),
+    postOrder: calculatePositionMargin({
+      adjustedMmf: adjustedMmf?.postOrder,
+      notionalTotal: notionalTotal?.postOrder,
+    }).toFixed(tickSizeDecimals ?? USD_DECIMALS),
+  };
+
+  const { freeCollateral, marginUsage } = useSelector(getSubaccount, shallowEqual) ?? {};
+
+  const formConfig =
+    marginAction === MarginAction.ADD
+      ? {
+          formLabel: stringGetter({ key: STRING_KEYS.ADDING }),
+          buttonLabel: stringGetter({ key: STRING_KEYS.ADD_MARGIN }),
+          inputReceiptItems: [
+            {
+              key: 'cross-free-collateral',
+              label: stringGetter({ key: STRING_KEYS.CROSS_FREE_COLLATERAL }),
+              value: (
+                <DiffOutput
```

Comment:
nit: since DiffOutput is repeated a lot, can code golf it to a function:
```
const createDiffOutputComponent = (item, outputType) => (
  <DiffOutput
    withDiff={!!item.postOrder && item.current !== item.postOrder}
    value={item.current}
    newValue={item.postOrder}
    type={outputType}
  />
);
```